src/pages/signin/SignInPage.js

```diff
@@ -0,0 +1,157 @@
+import React, {Component} from 'react';
+import {
+    SafeAreaView,
+    Text,
+    Image,
+    View,
+} from 'react-native';
+import PropTypes from 'prop-types';
+import _ from 'underscore';
+import {withOnyx} from 'react-native-onyx';
+import compose from '../../libs/compose';
+import {Redirect} from '../../libs/Router';
+import ROUTES from '../../ROUTES';
+import ONYXKEYS from '../../ONYXKEYS';
+import styles from '../../styles/styles';
+import logo from '../../../assets/images/expensify-logo-round.png';
+import CustomStatusBar from '../../components/CustomStatusBar';
+import updateUnread from '../../libs/UnreadIndicatorUpdater/updateUnread/index';
+import LoginForm from './LoginForm';
+import GithubUsernameForm from './GithubUsernameForm';
+import PasswordForm from './PasswordForm';
+import ResendValidationForm from './ResendValidationForm';
+
+const propTypes = {
+    /* Onyx Props */
+
+    // The details about the account that the user is signing in with
+    account: PropTypes.shape({
+        // Whether or not the account already exists
+        accountExists: PropTypes.bool,
+
+        // Whether or not there have been chat reports shared with this user
+        canAccessExpensifyCash: PropTypes.bool,
+    }),
+
+    // The credentials of the person signing in
+    credentials: PropTypes.shape({
+        login: PropTypes.string,
+        githubUsername: PropTypes.string,
+        password: PropTypes.string,
+        twoFactorAuthCode: PropTypes.string,
+    }),
+
+    // The session of the logged in person
+    session: PropTypes.shape({
+        // Error to display when there is a session error returned
+        error: PropTypes.string,
+    }),
+};
+
+const defaultProps = {
+    account: {},
+    session: null,
+    credentials: {},
+};
+
+class SignInPage extends Component {
+    componentDidMount() {
+        // Always reset the unread counter to zero on this page
+        updateUnread(0);
```

Comment:
This feels like it should happen somewhere else like the main page with all the routes on it.